lib/spack/spack/cmd/containerize.py

```diff
@@ -12,14 +12,41 @@
 level = "long"
 
 
+def setup_parser(subparser):
+    subparser.add_argument(
+        '-f', '--file', default=None,
+        dest='specfile', metavar='SPEC_YAML_FILE',
+        help="spack.yaml for which to generate the recipe. If not provided"
+             "try the env_dir provided by `spack -e`, if that also doesn't"
+             "exist try `./spack.yaml`.")
```

Comment:
The `-f` flag was originally present in  #14202 and removal was explicitly required, see https://github.com/spack/spack/pull/14202#discussion_r371650846 and the initial straw man implementation at  https://github.com/alalazo/spack-container/blob/master/container/cmd/containerize.py

Reply:
Do you know if @tgamblin had a deeper reason for this? I was planning on having all my environment definition files (spack.yaml) in a single location and checking out the ones I'm currently thinking about/wanting to build/etc.

Reply:
> Do you know if @tgamblin had a deeper reason for this?

For removing the `-f` flag? Initially the container configuration was in its own file and was either referencing or embedding an environment. You can have a good idea of how it looked like thinking that the `container` attribute was at the top level and the environment embedded under it. The file could be called with whatever name, so a `-f` option made sense.

When we decided to go for having the `container` configuration *be part* of Spack environment we didn't need a `-f` option anymore since the file is required to be called `spack.yaml`. An option to choose the directory location was already present (`spack -D ...`).

We went with the current design mainly to avoid increasing:
- The number of different configuration files
- The complexity for users

at the expense of a bit of modularity